substrate/client/sysinfo/src/sysinfo.rs

```diff
@@ -624,10 +669,20 @@ pub fn benchmark_sr25519_verify(limit: ExecutionLimit) -> Throughput {
 /// Optionally accepts a path to a `scratch_directory` to use to benchmark the
 /// disk. Also accepts the `requirements` for the hardware benchmark and a
 /// boolean to specify if the node is an authority.
-pub fn gather_hwbench(scratch_directory: Option<&Path>) -> HwBench {
+pub fn gather_hwbench(scratch_directory: Option<&Path>, requirements: &Requirements) -> HwBench {
+	let parallel_num_cores = requirements
+		.0
+		.iter()
+		.filter_map(|requirement| requirement.metric.num_cores())
+		.next()
+		.unwrap_or(1);
 	#[allow(unused_mut)]
 	let mut hwbench = HwBench {
 		cpu_hashrate_score: benchmark_cpu(DEFAULT_CPU_EXECUTION_LIMIT),
+		parallel_cpu_hashrate_score: benchmark_cpu_parallelism(
+			DEFAULT_CPU_EXECUTION_LIMIT,
+			parallel_num_cores,
+		),
```

Comment:
I think a better way to do this would be something like this:

```rust
let cpu_hashrate_score = benchmark_cpu(DEFAULT_CPU_EXECUTION_LIMIT);
let parallel_cpu_hashrate_score = requirements.0.filter_map(|req| {
    if let Self::Blake2256Parallel { num_cores } = req {
        benchmark_cpu_parallelism(DEFAULT_CPU_EXECUTION_LIMIT, num_cores)
    } else {
        None
    }
}).next().unwrap_or(cpu_hashrate_score);
```

In case the parallel requirement is missing it won't run the benchmark twice.

Reply:
Done.